### ISSUE TITLE
[C#] Implement new Concept Exercise: inheritance

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/v3#968]
This issue describes how to implement the `inheritance` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the Concept of Inheritance in C#.

## Learning objectives

- Know what inheritance is.
- Know how to inherit from a class.
- Know that all types inherit from `object`.
- Know what abstract and sealed classes are.
- Know what abstract and virtual methods are.
- Know how to override methods.
- Know about the `protected` visibility modifier.

## Out of scope

- Extending types through extension methods.

## Concepts

This Concepts Exercise's Concepts are:

- `inheritance`: know what inheritance is; know how to inherit from a class; know that all types inherit from `object`; know what abstract and sealed classes are; know what abstract and virtual methods re; know how to override methods; know about the `protected` visibility modifier.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `classes`: defining base and derived classes.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Inheritance][inheritance]: tutorial introducing the various aspects of inheritance.

### After

- [Inheritance][inheritance]: tutorial introducing the various aspects of inheritance.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-doc]: https://github.com/exercism/v3/blob/master/reference/concepts/inheritance.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[inheritance]: https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/inheritance

**Comment by ErikSchierboom** [comment]
I'm working on this.
